pal/system-info/web/system-info.ts

```diff
@@ -167,11 +167,25 @@ class SystemInfo extends EventTarget {
                 imageBitmap?.close();
             }).catch((err) => {});
         }
-        this.supportCapability = {
-            webp: supportWebp,
-            gl: supportWebGL,
-            canvas: supportCanvas,
-            imageBitmap: supportImageBitmap,
+
+        let supportWebGL2 = (!!window.WebGL2RenderingContext);
+        const userAgent = window.navigator.userAgent.toLowerCase();
+        if (userAgent.indexOf('safari') !== -1 && userAgent.indexOf('chrome') === -1
+            || this.browserType === BrowserType.UC // UC browser implementation doesn't conform to WebGL2 standard
+        ) {
+            supportWebGL2 = false;
+        }
+        this._featureMap = {
+            [Feature.CANVAS]: supportCanvas,
+            [Feature.GL]: supportWebGL,
+            [Feature.WEBGL2]: supportWebGL2,
+            [Feature.WEBP]: supportWebp,
+            [Feature.IMAGE_BIT_MAP]: supportImageBitmap,
+
+            [Feature.INPUT_KEYBOARD]: document.documentElement.onkeyup !== undefined,
+            [Feature.INPUT_MOUSE]: !EDITOR && document.documentElement.onmouseup !== undefined,
+            [Feature.INPUT_TOUCH]: !!(document.documentElement.ontouchstart !== undefined || document.ontouchstart !== undefined || navigator.msPointerEnabled),
```

Comment:
Same as previous suggestion, please set to true or check both mouse and touch event to here